lint_test/prefer_early_return_test.dart

```diff
@@ -0,0 +1,237 @@
+// ignore_for_file: dead_code, cyclomatic_complexity, no_equal_then_else, prefer_match_file_name
+
+// ignore: no_empty_block
+void _doSomething() {}
+
+void oneIf() {
+  //expect_lint: prefer_early_return
+  if (true) {
+    _doSomething();
+  }
+}
+
+int oneIfWithReturnValue() {
+  //no lint
+  if (true) {
+    _doSomething();
+  }
+
+  return 1;
+}
+
+void oneIfWithReturn() {
+  //expect_lint: prefer_early_return
+  if (true) {
+    _doSomething();
+  }
+
+  return;
+}
+
+void nestedIf2() {
+  //expect_lint: prefer_early_return
+  if (true) {
+    if (true) {
```

Comment:
Can we show a warning here as well?

Reply:
this block is illuminated by the warning in its entirety